pilot/pkg/model/authentication.go

```diff
@@ -55,11 +68,53 @@ func GetConsolidateAuthenticationPolicy(store IstioConfigStore, service *Service
 }
 
 // ConstructSdsSecretConfig constructs SDS Sececret Configuration.
-func ConstructSdsSecretConfig(name string, sdsUdsPath string) *auth.SdsSecretConfig {
+func ConstructSdsSecretConfig(name, sdsUdsPath, tokenMountFileName string) *auth.SdsSecretConfig {
 	if name == "" || sdsUdsPath == "" {
 		return nil
 	}
 
+	gRPCConfig := &core.GrpcService_GoogleGrpc{
+		TargetUri:  sdsUdsPath,
+		StatPrefix: SDSStatPrefix,
+		ChannelCredentials: &core.GrpcService_GoogleGrpc_ChannelCredentials{
+			CredentialSpecifier: &core.GrpcService_GoogleGrpc_ChannelCredentials_LocalCredentials{
+				LocalCredentials: &core.GrpcService_GoogleGrpc_GoogleLocalCredentials{},
+			},
+		},
+		CallCredentials: []*core.GrpcService_GoogleGrpc_CallCredentials{
+			&core.GrpcService_GoogleGrpc_CallCredentials{
+				CredentialSpecifier: &core.GrpcService_GoogleGrpc_CallCredentials_GoogleComputeEngine{
+					GoogleComputeEngine: &types.Empty{},
+				},
+			},
+		},
+	}
+
+	if _, err := os.Stat(tokenMountFileName); err == nil {
+		// token volume mount file for k8s sa jwt.
+		tokenMountonfig := &v2alpha.FileBasedMetadataConfig{
+			SecretData: &core.DataSource{
+				Specifier: &core.DataSource_Filename{
+					Filename: tokenMountFileName,
+				},
+			},
+
+			HeaderKey: k8sSAJwtTokenHeaderKey,
+		}
+
+		callCred := &core.GrpcService_GoogleGrpc_CallCredentials{
+			CredentialSpecifier: &core.GrpcService_GoogleGrpc_CallCredentials_FromPlugin{
+				FromPlugin: &core.GrpcService_GoogleGrpc_CallCredentials_MetadataCredentialsFromPlugin{
+					Name:   fileBasedMetadataPlugName,
+					Config: protoToStruct(tokenMountonfig),
+				},
+			},
+		}
+
+		gRPCConfig.CredentialsFactoryName = fileBasedMetadataPlugName
```

Comment:
As long as this CredentialsFactoryName is set to "envoy.grpc_credentials.file_based_metadata", Envoy only fetches call credentials from via from_plugin. 
https://github.com/envoyproxy/envoy/blob/570a403b8882ea34309bbe73209fee41417fa10e/source/common/grpc/google_grpc_creds_impl.cc#L136
Maybe add a comment to clarify that there is no fallback to fetching call creds from GCE.

Reply:
added comment.

Reply:
nit: If the `ChannelCredentials` is not used in this case, it's better to only set it in the else part, so it will be very clear even without a comment.

Reply:
channelcred is used in both, if/else here is only for callcred

Reply:
Thanks, I understand Jimmy's comment wrongly here.